packages/textfield/README.md

```diff
@@ -0,0 +1,267 @@
+# `<mwc-textfield>` & `<mwc-textarea>` [![Published on npm](https://img.shields.io/npm/v/@material/mwc-textfield.svg)](https://www.npmjs.com/package/@material/mwc-textfield)
+
+> IMPORTANT: The Material Web Components are a work in progress and subject to
+> major changes until 1.0 release.
+
+Textfields let users enter and edit text.
+
+<img src="images/action.png" width="28.5714%">
+
+[Material Design Guidelines: textfields](https://material.io/design/components/text-fields.html)
+
+## Installation
+
+```sh
+npm install @material/mwc-textfield
+```
+
+> NOTE: The Material Web Components are distributed as ES2017 JavaScript
+> Modules, and use the Custom Elements API. They are compatible with all modern
+> browsers including Chrome, Firefox, Safari, Edge, and IE11, but an additional
+> tooling step is required to resolve *bare module specifiers*, as well as
+> transpilation and polyfills for Edge and IE11. See
+> [here](https://github.com/material-components/material-components-web-components#quick-start)
+> for detailed instructions.
+
+## Example usage
+
+### Standard / Filled
+
+<img src="images/standard.png" width="28.5714%">
+
+```html
+<mwc-textfield label="My Textfield"></mwc-textfield>
+
+<script type="module">
+  import '@material/mwc-textfield';
+</script>
+```
+
+### Icon - Leading
+
+<img src="images/icon-leading.png" width="28.5714%">
+
+```html
+<mwc-textfield label="My Textfield" icon="event"></mwc-textfield>
+
+<script type="module">
+  import '@material/mwc-textfield';
+  import '@material/mwc-icon/mwc-icon-font.js';
+</script>
+```
+
+### Icon - Trailing
+
+<img src="images/icon-trailing.png" width="28.5714%">
+
+```html
+<mwc-textfield label="My Textfield" iconTrailing="delete"></mwc-textfield>
+
+<script type="module">
+  import '@material/mwc-textfield';
+  import '@material/mwc-icon/mwc-icon-font.js';
+</script>
+```
+
+### Helper Text
+
+<img src="images/helper.png" width="28.5714%">
+
+```html
+<mwc-textfield label="My Textfield" helper="Helper Text"></mwc-textfield>
+
+<script type="module">
+  import '@material/mwc-textfield';
+</script>
+```
+
+### Primary Color
+
+<img src="images/color-primary.png" width="28.5714%">
+
+```html
+<style>
+  mwc-textfield {
+    --mdc-theme-primary: green;
+  }
+</style>
+
+<mwc-textfield
+    label="My Textfield"
+    iconTrailing="delete"
+    required>
+</mwc-textfield>
+
+<script type="module">
+  import '@material/mwc-textfield';
+  import '@material/mwc-icon/mwc-icon-font.js';
+</script>
+```
+
+### Error Color
+
+<img src="images/color-error.png" width="28.5714%">
+
+```html
+<style>
+  mwc-textfield {
+    --mdc-theme-error: blue;
```

Comment:
Is there a default error color? Maybe something red is a better example?

Also this screenshot looks different to all the others because it doesn't have a floating label, which I find slightly jarring (since the thing we're highlighting here is how errors work, not how floating labels work).

Maybe copy the example from the spec (https://material.io/design/components/text-fields.html#anatomy scroll to "Assitive elements") to show errors?

It might also be nice to combine the Primary Color and Error Color examples into a single "Custom colors" that sets both?

Reply:
There is a default error color, I was trying to show that it is customizable. Should I choose something different but still reddish like `FireBrick` or `Salmon`?

Reply:
Your call. TBH I just think it looks kinda ugly. I think the examples here also serve secondarily as a gallery of what people can expect their app to look like if they use the components.

You might not even want to use error text as the example of a custom color, you could just use a regular input label, which seems like something a user is more likely to actually want to customize.

Reply:
removed